frontend/test/metabase-lib/lib/queries/StructuredQuery.unit.spec.js

```diff
@@ -296,6 +328,96 @@ describe("StructuredQuery", () => {
         });
       });
     });
+
+    describe("aggregationOperators", () => {
+      it("has the same aggregation operators when query have no custom fields", () => {
+        expect(query.aggregationOperators()).toEqual(
+          query.table().aggregationOperators(),
+        );
+        expect(
+          query.aggregationOperators().map(aggregation => aggregation.short),
+        ).toEqual([
+          "rows",
+          "count",
+          "sum",
+          "avg",
+          "median",
+          "distinct",
+          "cum-sum",
+          "cum-count",
+          "stddev",
+          "min",
+          "max",
+        ]);
+      });
+
+      it("only show aggregation operators on available fields", () => {
+        const queryWithoutNumericFields = makeQueryWithoutNumericFields();
+
+        expect(queryWithoutNumericFields.aggregationOperators()).toEqual(
+          queryWithoutNumericFields.table().aggregationOperators(),
+        );
+        expect(
+          queryWithoutNumericFields
+            .aggregationOperators()
+            .map(aggregation => aggregation.short),
+        ).toEqual(["rows", "count", "distinct", "cum-count", "min", "max"]);
+      });
+
+      it("shows `avg` aggregation when having a numeric custom field on table without numeric fields", () => {
+        const query = makeQueryWithoutNumericFields().addExpression(
+          "custom_numeric_field",
+          // Expression: case([ID] = 1, 11, 99)
+          ["case", [["=", ORDERS.ID, 1], 11], { default: 99 }],
+        );
+
+        expect(query.aggregationOperators()).not.toEqual(
+          query.table().aggregationOperators(),
+        );
+        expect(
+          query.aggregationOperators().map(aggregation => aggregation.short),
+        ).toEqual([
```

Comment:
This asserts that previously we ensure there wouldn't be aggregations for numeric fields here (previous test), but after adding an expression that would result in a numeric field, this aggregations for numeric fields start being available.